go/kbfs/search/indexer.go

```diff
@@ -513,6 +530,37 @@ func (i *Indexer) updateChild(
 	return nil
 }
 
+func (i *Indexer) renameChild(
+	ctx context.Context, parentNode libkbfs.Node, parentDocID string,
+	childName data.PathPartString, revision kbfsmd.Revision) (err error) {
+	ptr, n, ei, err := i.getCurrentPtrAndNode(ctx, parentNode, childName)
+	if err != nil {
+		return err
+	}
+
+	if i.blocksDb == nil {
+		return errors.New("No indexed blocks db")
+	}
+
+	// Get the docID.
+	_, docID, err := i.blocksDb.Get(ctx, ptr)
+	if err != nil {
+		// Treat "not found" errors as real errors, since a rename
+		// implies that the doc should have already been indexed.
```

Comment:
What if it's renamed before content index is done?

Reply:
The indexing is pegged at a particular revision, and won't be updated as renames happen.  Once the revision is fully indexed, then it will (in a future PR) move onto processing updates.